backends/aoti/runtime/AotiBackend.cpp

```diff
@@ -0,0 +1,532 @@
+    AOTITensorHandle tensor,
+    int64_t** ret_strides) {
+    if (tensor_to_strides.find(tensor) == tensor_to_strides.end()) {
+      std::vector<int64_t> strides(tensor->dim());
+      for (int i = 0; i < tensor->dim(); i++) {
+        strides[i] = tensor->strides()[i];
+      }
+      tensor_to_strides[tensor] = strides;
+    }
+    *ret_strides = tensor_to_strides[tensor].data();
+    std::cout << "getting strides from tensor " << tensor << " with dim " << tensor->dim() << std::endl;
+    for (int i = 0; i < tensor->dim(); i++) {
+      std::cout << "strides " << i << " = " << *ret_strides[i] << std::endl;
+    }
+    return Error::Ok;
+  }
+
+  AOTITorchError aoti_torch_get_dtype(
+    AOTITensorHandle tensor,
+    int32_t* ret_dtype) {
+    *ret_dtype = static_cast<int32_t>(tensor->scalar_type());
+    return Error::Ok;
+  }
+
+  AOTITorchError aoti_torch_get_sizes(
+    AOTITensorHandle tensor,
+    int64_t** ret_sizes) {
+    if (tensor_to_sizes.find(tensor) == tensor_to_sizes.end()) {
+      std::vector<int64_t> sizes(tensor->dim());
+      for (int i = 0; i < tensor->dim(); i++) {
+        sizes[i] = tensor->size(i);
+      }
+      tensor_to_sizes[tensor] = sizes;
+    }
+    *ret_sizes = tensor_to_sizes[tensor].data();
+    std::cout << "getting sizes from tensor " << tensor << " with dim " << tensor->dim() << std::endl;
+    for (int i = 0; i < tensor->dim(); i++) {
+      std::cout << "size " << i << " = " << *ret_sizes[i] << std::endl;
+    }
+    return Error::Ok;
+  }
+
+  AOTITorchError aoti_torch_get_storage_size(
+    AOTITensorHandle tensor,
+    int64_t* ret_size) {
+    throw std::runtime_error("Cannot get storage size on ETensor");
+  }
+  AOTITorchError aoti_torch_create_tensor_from_blob_v2(
+    void* data,
+    int64_t ndim,
+    const int64_t* sizes_ptr,
+    const int64_t* strides_ptr,
+    int64_t storage_offset,
+    int32_t dtype,
+    int32_t device_type,
+    int32_t device_index,
+    AOTITensorHandle* ret_new_tensor,
+    int32_t layout,
+    const uint8_t* opaque_metadata,
+    int64_t opaque_metadata_size) {
+    throw std::runtime_error("Not creating Tensor from blob here");
+  }
+  AOTITorchError aoti_torch_create_cuda_stream_guard(
+    void* stream,
+    int32_t device_index,
+    CUDAStreamGuardHandle* ret_guard) {
+    std::cout<<"Entering stream guard for device " << device_index << std::endl;
+    return Error::Ok;
+  }
+  AOTITorchError aoti_torch_delete_cuda_stream_guard(
+    CUDAStreamGuardHandle guard) {
+    std::cout<<"Exiting stream guard" << std::endl;
+    return Error::Ok;
+  }
+  int aoti_torch_device_type_cpu() {
+    // Let's say cpu is 0 for ET as well
+    return 0;
+  }
+  __attribute__((__visibility__("default"))) int32_t aoti_torch_device_type_cuda() {
+    // Let's say cuda is 1 for ET as well
+    return 1;
+  }
+  __attribute__((__visibility__("default"))) int32_t aoti_torch_dtype_float32() {
+    // Let assume the dtype here is all we will support
+    return 6;
+  }
```

Comment:
since this is a generic AOTI delegate can we put a TODO here to support other dtypes so it'll show up if we grep? also why can't we use the ExecuTorch ScalarType enum?